infra/core/config/configstore.bicep

```diff
@@ -15,8 +15,6 @@ param keyValueNames array = []
 @description('Specifies the values of the key-value resources.')
 param keyValueValues array = []
 
-@description('The principal ID to grant access to the Azure App Configuration store')
-param principalId string
```

Comment:
Why remove principal Id?  Our design allows the user to run the application locally with their principal Id, so I think by removing this the user will no longer be able to run it locally.

Reply:
This rbac role assignment is raised up to main.bicep because container created with identity. The sequence will be:
1. App Config is created
2. Container app is created with App Config's endpoint address filled in env variable
3. This Rbac role assignment is created.
This is to avoid dependency on principal Id before the managed identity is created